src/providers/ldap/ldap_child.c

```diff
@@ -279,6 +279,7 @@ static errno_t select_principal_from_keytab(TALLOC_CTX *mem_ctx,
 
     kerr = sss_krb5_init_context(&krb_ctx);
     if (kerr) {
+        krb_ctx = NULL;
```

Comment:
I'm surprised underlying `krb5_init_context()` touches 'krb_ctx' in case of failure.

Wouldn't it be more consistent to fix `sss_krb5_init_context()` instead?

Reply:
Btw, such a change would apply both to 2-10 and 2-9.
But error printing still requires different patches...

Reply:
Hi,

I modified `sss_krb5_init_context()` to make sure the context is only touched if `krb5_init_context()` was successful.

bye,
Sumit